src/formio/components/SoftRequiredErrors.js

```diff
@@ -33,7 +33,11 @@ class SoftRequiredErrors extends FormioContentField {
     // check which components have an empty value
     for (const component of softRequiredComponents) {
       const isEmpty = component.isEmpty();
-      if (isEmpty) missingFieldLabels.push(component.label);
+      const isParentVisible = component.parent?.visible;
```

Comment:
what it the grand parent is hidden?

Reply:
Yes, you are right I didn't think about it.

Reply:
After some research I saw that even if we have nested components this works. If, as you mentioned, the grand parent is hidden, all the children are hidden as well. I understood that formio (the UI) does not render the children of a hidden component, even if the json configuration has `hidden=false`.

Reply:
That's correct - I just didn't know if `component.visible` took into account is parents or not :)